packages/flutter/lib/src/material/radio.dart

```diff
@@ -75,22 +367,17 @@ const double _kInnerRadius = 4.5;
 class Radio<T> extends StatefulWidget {
   /// Creates a Material Design radio button.
   ///
-  /// The radio button itself does not maintain any state. Instead, when the
-  /// radio button is selected, the widget calls the [onChanged] callback. Most
-  /// widgets that use a radio button will listen for the [onChanged] callback
-  /// and rebuild the radio button with a new [groupValue] to update the visual
-  /// appearance of the radio button.
+  /// {@macro flutter.material.Radio.groupValue}
   ///
   /// The following arguments are required:
   ///
-  /// * [value] and [groupValue] together determine whether the radio button is
-  ///   selected.
+  /// * [value] is used to identify this radio
   /// * [onChanged] is called when the user selects this radio button.
   const Radio({
     super.key,
     required this.value,
-    required this.groupValue,
```

Comment:
For where we assert against the new and old API choice, I think some of them are not yet considering that both `groupValue` and `onChanged` are required for the old path - but I did not look too closely since this is just an initial pass. We can check later.